intake_stac/catalog.py

```diff
@@ -398,12 +444,18 @@ def _get_driver(self, entry):
             # 'application/geopackage+sqlite3': 'geopandas',
             'application/geo+json': 'geopandas',
         }
+
         entry_type = entry.get('type', NULL_TYPE)
 
         if entry_type is NULL_TYPE:
-            warnings.warn(
-                f'TODO: handle case with entry without type field. This entry was: {entry}'
-            )
+            # Fallback to common file suffix mappings
+            suffix = os.path.splitext(entry['href'])[-1]
+            if suffix in ['.nc', '.h5', '.hdf']:
+                entry_type = 'application/netcdf'
+            # produces excess output for large catalogs
+            #    warnings.warn(
+            #        f'TODO: handle case with entry without type field. This entry was: {entry}'
+            #    )
```

Comment:
I think we should leave this warning in place. Perhaps adding some docs that show users how to set the warning filter, e.g.

```python
 warnings.simplefilter('once')
```

I think this warning is particularly important without a mime type, intake has no way of opening the asset. We need to tell the user that up front.

Reply:
Sounds good. Once a user sees that warning, is there a way to manually specify the mimetype after reading the STAC catalog? That would be a nice solution and better warning message to display.

Reply:
actually @jhamman - `warnings.simplefilter('once')` doesn't work with the current implementation, because it is already raised once per item. Could move to the catalog level and warn 'some items did not have type assigned...' For now I'll put in a `warnings.simplefilter('ignore') with a comment at the top of example notebooks, but leave the current implementation.